### ISSUE TITLE
F841: `unused-variable` conflicts with Pydantic's ForwardRef

### ISSUE THREAD

**Issue by Konfus-ius** [astral-sh/ruff#18460]
### Summary

Taking the following class definition

```python
from pydantic import BaseModel, ConfigDict, field_validator

class ClassName(BaseModel):
    output: ForwardRef("unused_variable")

    @classmethod
    def initialize(cls, db_repository: DatabaseRepository) -> "WhatEver":
        unused_variable = List[Literal[*output]] 
        cls.model_rebuild()
        return cls
```
ruff check shows a finding for F841 unused variable. This is theroretically correct but actually wrong. As the pydantic ForwardRef makes use of the `unused_variable`.

Request:
 - Either "flexibilize" F841 such that pydantic ForwardRef is properly taken into account or
 - add rule to enforce getting rid of ForwardRef within pydantic (if wanted)

Thanks for guidance

**Comment by Viicos** [comment]
Not a Ruff maintainer but I don't think special casing this is necessary. `model_rebuild()` fetches the current namespace where it is called to resolve forward annotations, which is really uncommon behavior (and only works with some Python implementations). Instead, we recommend explicitly providing the rebuild namespace with `model_rebuild(_types_namespace={'unused_variable': unused_variable})`.

**Comment by Konfus-ius** [comment]
Hi @Viicos 

thanks for the feedback, we will check this approach. In best case, we get fully rid of the ForwardRefs ...

But your approach is much better and is in line with linting.
